File: server/java/src/main/java/com/onlinepayments/example/service/interfaces/clients/HostedCheckoutClientInterface.java
package com.onlinepayments.example.service.interfaces.clients;

import com.onlinepayments.example.service.dtos.hostedcheckout.CreateHostedCheckoutRequestDto;
import com.onlinepayments.example.service.dtos.hostedcheckout.CreateHostedCheckoutResponseDto;
import com.onlinepayments.example.service.dtos.paymentbyhostedcheckoutid.GetPaymentByHostedCheckoutIdResponseDto;

public interface HostedCheckoutClientInterface {
    CreateHostedCheckoutResponseDto createHostedCheckout(CreateHostedCheckoutRequestDto request);

    GetPaymentByHostedCheckoutIdResponseDto getPaymentByHostedCheckoutId(String id);
}

File: server/nodejs/src/infrastructure/interfaces/hosted-checkout-client-interface.ts
import { FastifyBaseLogger } from 'fastify';
import { CreateHostedCheckoutResponseDto } from '../../business/dtos/hosted-checkout/create-hosted-checkout-response-dto';
import { CreateHostedCheckoutRequestDto } from '../../business/dtos/hosted-checkout/create-hosted-checkout-request-dto';
import { GetPaymentByHostedCheckoutResponseDto } from '../../business/dtos/hosted-checkout/get-payment-by-hosted-checkout-response-dto';

export interface IHostedCheckoutClient {
    createHostedCheckout(
        request: CreateHostedCheckoutRequestDto,
        logger: FastifyBaseLogger,
    ): Promise<CreateHostedCheckoutResponseDto>;

    getPaymentByHostedCheckoutId(
        hostedCheckoutId: string,
        logger: FastifyBaseLogger,
    ): Promise<GetPaymentByHostedCheckoutResponseDto | null>;
}
